### ISSUE TITLE
Snap Camera no longer available as an input source for any app, including Zoom

### ISSUE THREAD

**Issue by Hazel4552** [jaku/SnapCameraPreservation#87]
I have a Macbook Pro with Intel Core i7, running Sonoma 14.4. When selecting the Snap Camera input on Zoom, it abruptly began showing a black screen after working fine up to this point. I tried reinstalling both Zoom and Snap Camera -- now Snap Cam is not available as an input at all. The app itself works fine along with all the filters, but does not display as an available input for any of the apps I've tried, including Zoom, Google Meet, and OBS. I've checked all my system preferences and do not believe that they are the issue.

**Comment by AbdullahAlbahri** [comment]
same, I think servers are down

**Comment by kineticscreen** [comment]
You probably just need to run the Terminal command again since recently installing a MacOS update ☺️

**Comment by Hazel4552** [comment]
@kineticscreen  That was the first thing I tried. Didn't work.

I hadn't yet installed the latest update when the input source broke. I installed it after to see if it would help. Before installing the update, SnapCamera appeared as in input in Zoom, but showed a black screen, similar to what users describe in another open ticket. After the update, the input option disappeared altogether from all apps.

Any other suggestions?

**Comment by Hazel4552** [comment]
@AbdullahAlbahri you can host it locally if you want. I'm just not technical enough to figure out how to install it. Check out the [project by Ptrumpis](https://github.com/ptrumpis/snap-camera-server).

**Comment by kineticscreen** [comment]
I assume you're aware of the recent changes to MacOS regarding the Virtual Webcams, and the need to do that override trick to unlock the previous functionality?

**Comment by Hazel4552** [comment]
@kineticscreen Are you referring to [this](https://support.apple.com/en-us/108387)? When I tried it, my Mac gave me an error. Going to try again.

**Comment by kineticscreen** [comment]
Yup, you have to get that working or the 'virtual webcam' won't work.

And just FYI even though I have got that working fine, after doing the latest MacOS update I was again just getting a black input, so had to run the terminal command / install again and that fixed it. So it might be that OS updates rewquire it to be reinstalled. Although thankfully the legacy video support wasn't affected.

**Comment by Hazel4552** [comment]
Thank you so much @kineticscreen

**Comment by Hazel4552** [comment]
@kineticscreen there are three options for recovery mode and the instructions don't specify which one to use. Would you happen to know?

Command-R: Start up from the built-in macOS Recovery System. Use this key combination to reinstall the latest macOS that was installed on your system, or to use the other apps in macOS Recovery.

Option-Command-R: Start up from macOS Recovery over the internet. Use this key combination to reinstall macOS and upgrade to the latest version of macOS that’s compatible with your Mac.

Option-Shift-Command-R: Start up from macOS Recovery over the internet. Use this key combination to reinstall the version of macOS that came with your Mac or the closest version that’s still available.

**Comment by Onanonad** [comment]
I've got the snapcam working great up until an hour ago, been trying everything to fix it and still come up black or yellow in any app even with obs. :(

**Comment by rudierai** [comment]
> I've got the snapcam working great up until an hour ago, been trying everything to fix it and still come up black or yellow in any app even with obs. :(

Having this same issue too. Yellow screen on OBS and running Mac Ventura. Worked fine before so unsure what may have changed

**Comment by kineticscreen** [comment]
I just had this happen to me too - but yet again, just re-running the original Terminal patch (and it reinstalling the Snap Camera app) fixed it. And I'm running OBS 30.1.0 and today's 14.4.1 macOS update.

**Comment by rudierai** [comment]
> I just had this happen to me too - but yet again, just re-running the original Terminal patch (and it reinstalling the Snap Camera app) fixed it. And I'm running OBS 30.1.0 and today's 14.4.1 macOS update.

Please can you expand on what you did here? Did you delete snap camera and start from the beginning? Did you have to enter the override command in the terminal (this doesn’t work for me and says command not found—and I’m still using Ventura anyway)? Any extra help would be great, thanks!

**Comment by Onanonad** [comment]
Ok, so it's working again. What I did is when it's pixelated I'd restart my macbook. Sometimes it helps, and if it doesn't I'd simply delete the snapcam app, and run the terminal code again to patch the app, and it'll work again. But you're gonna have to do this again and again because apparently the problem happens quite regularly.

**Comment by rudierai** [comment]
> Ok, so it's working again. What I did is when it's pixelated I'd restart my macbook. Sometimes it helps, and if it doesn't I'd simply delete the snapcam app, and run the terminal code again to patch the app, and it'll work again. But you're gonna have to do this again and again because apparently the problem happens quite regularly.

This is perfect, thank you! Thankfully I was able to be able to repatch the app without deleting it and it seems to be working now!

**Comment by kineticscreen** [comment]
I haven't properly confirmed this, but it may be that this problem occurs whenever the computer is rebooted?

**Comment by Onanonad** [comment]
Anybody else has problems with snapcam again?

**Comment by marinarobbins** [comment]
Just a heads up that I have followed all the steps relative to override/plug-ins and the patch. All is well except snap camera freezes when using another app like Teams. Hopefully this will be fixed. Thank you for all you are doing!
